playground/backend/internal/setup_tools/builder/setup_builder.go

```diff
@@ -31,73 +31,139 @@ const (
 	prep, err := utils.GetPreparators(sdk, paths.AbsoluteSourceFilePath)
 	if err != nil {
 		return nil, err
 	}
-	executorConfig := sdkEnv.ExecutorConfig
 	builder := executors.NewExecutorBuilder().
-		WithExecutableFileName(paths.AbsoluteExecutableFilePath).
-		WithWorkingDir(paths.AbsoluteBaseFolderPath).
-		WithValidator().
-		WithSdkValidators(val).
 		WithPreparator().
 		WithSdkPreparators(prep).
+		ExecutorBuilder
+	return &builder, err
+}
+
+// Compiler return executor with set args for compiler
+func Compiler(paths *fs_tool.LifeCyclePaths, sdkEnv *environment.BeamEnvs) *executors.ExecutorBuilder {
+	sdk := sdkEnv.ApacheBeamSdk
+	executorConfig := sdkEnv.ExecutorConfig
+	builder := executors.NewExecutorBuilder().
 		WithCompiler().
 		WithCommand(executorConfig.CompileCmd).
+		WithWorkingDir(paths.AbsoluteBaseFolderPath).
 		WithArgs(executorConfig.CompileArgs).
 		WithFileName(paths.AbsoluteSourceFilePath).
+		ExecutorBuilder
+
+	switch sdk {
+	case pb.Sdk_SDK_JAVA:
+		builder.
+			WithCompiler().
+			WithFileName(GetFirstFileFromFolder(paths.AbsoluteSourceFileFolderPath))
+	}
+	return &builder
+}
+
+// Runner return executor with set args for runner
+func Runner(paths *fs_tool.LifeCyclePaths, pipelineOptions string, sdkEnv *environment.BeamEnvs) (*executors.ExecutorBuilder, error) {
+	sdk := sdkEnv.ApacheBeamSdk
+
+	if sdk == pb.Sdk_SDK_JAVA {
+		pipelineOptions = utils.ReplaceSpacesWithEquals(pipelineOptions)
+	}
+	executorConfig := sdkEnv.ExecutorConfig
+	builder := executors.NewExecutorBuilder().
 		WithRunner().
+		WithWorkingDir(paths.AbsoluteBaseFolderPath).
 		WithCommand(executorConfig.RunCmd).
 		WithArgs(executorConfig.RunArgs).
 		WithPipelineOptions(strings.Split(pipelineOptions, " ")).
-		WithTestRunner().
-		WithCommand(executorConfig.TestCmd).
-		WithArgs(executorConfig.TestArgs).
-		WithWorkingDir(paths.AbsoluteSourceFileFolderPath).
 		ExecutorBuilder
 
 	switch sdk {
-	case pb.Sdk_SDK_JAVA: // Executable name for java class will be known after compilation
-		args := make([]string, 0)
-		for _, arg := range executorConfig.RunArgs {
-			if strings.Contains(arg, javaLogConfigFilePlaceholder) {
-				logConfigFilePath := filepath.Join(paths.AbsoluteBaseFolderPath, javaLogConfigFileName)
-				arg = strings.Replace(arg, javaLogConfigFilePlaceholder, logConfigFilePath, 1)
-			}
-			args = append(args, arg)
+	case pb.Sdk_SDK_JAVA: // Executable name for java class is known after compilation
+		args := replaceLogPlaceholder(paths, executorConfig)
+		className, err := paths.ExecutableName(paths.AbsoluteExecutableFileFolderPath)
+		if err != nil {
+			return nil, fmt.Errorf("no executable file name found for JAVA pipeline at %s", paths.AbsoluteExecutableFileFolderPath)
 		}
-		builder = builder.WithRunner().WithArgs(args).ExecutorBuilder
-		builder = builder.WithTestRunner().WithWorkingDir(paths.AbsoluteBaseFolderPath).ExecutorBuilder //change directory for unit test
+		builder = builder.
+			WithRunner().
```

Comment:
We already have `WithRunner()` at the beginning of the building of the executor, should we call it once more here?

Reply:
Yes